### DISCUSSION TITLE
[Proposal]: System.String.RSplit

### DISCUSSION THREAD

**Discussion by mikeronayne** [dotnet/csharplang#8558] [Language Ideas]
This should behave exactly like the [System.String.Split](https://learn.microsoft.com/en-us/dotnet/api/system.string.split?view=net-8.0) overloads that accept `count`, except we start counting delimiters from the _right_ instead of the _left_.

Python exposes this as [rsplit](https://docs.python.org/3/library/stdtypes.html#str.rsplit) (tests showing behavior [here](https://github.com/python/cpython/blob/83ba8c2bba834c0b92de669cac16fcda17485e0e/Lib/test/test_capi/test_unicode.py#L1110-L1135)). (Note that Python's `maxsplit` parameter has a bit different semantics than C#'s `count`.)

Alternatively, instead of adding a new `RSplit` method, I could see this also being implemented as a new `StringSplitOptions`.

**Comment by timcassell** [comment]
API requests go in [dotnet/runtime](https://github.com/dotnet/runtime/issues)

**Reply by mikeronayne** [reply]
Apologies. Moved: https://github.com/dotnet/runtime/issues/109583
